os/Makefile.unix

```diff
@@ -543,7 +543,7 @@ pass1dep: context tools/mkdeps$(HOSTEXEEXT) tools/compression/mkcompresstool.sh
 		$(MAKE) -C $$dir TOPDIR="$(TOPDIR)" EXTDIR="$(EXTDIR)" depend ; \
 	done
 
-pass2dep: context tools/mkdeps$(HOSTEXEEXT)
+pass2dep: context tools/mkdeps$(HOSTEXEEXT) smartfs
```

Comment:
In my opinion, we don't need to make a smartfs image at every build. I think we had better make it with new specific command like `make smartfs`. To get smartfs binary, changing config is inconvenient and it causes full build. Could you let me know your opinion, @yashwanth686007 @sijupunnnoose ?

Reply:
@sunghan-chang, this is same as romfs image creation. We cannot call "make smartfs" since it is not arm gcc build, it is host build. Instead we can call the mksmartfs.sh which will create the binary, but user has to do this manually. Advantage here in this case is nxfuse tool and binary would be created as part of build, developer need not do it manually. I would recommend the current approach, if you feel it is not right we can modify it, but i don't think it is correct.

Reply:
With the current approach, smarts image is created at every build. However smartfs may contain user contents, therefore we should avoid overwriting it. Also, we generally don't make frequent modifications in smartfs, so building it everytime is not necessary.

Reply:
@NitishAmbastha ,Yes, I agree with you. Only when CONFIG_SMARTFS_IMAGE is enabled, smartfs image will be created and downloaded/overwritten onto the flash. Also, I have followed pattern similar to romfs image creation procedure. As @sijupunnnoose  said, we can separate out smartfs image preparation outside normal build, but in this case, user needs to manually call the script.

Reply:
The sole point is that we run a risk of overwriting user content, by enabling config which is much easier option. In case of romfs it is not the same case.

Reply:
Ok. I have modified as discussed. 
For preparation of smartfs image: "make smartfs"
For downloading of image  : "make download userfs"
Please review.